packages/x-adapter/README.md

```diff
@@ -1,13 +1,218 @@
+  requestMapper: searchRequestMapper,
+  responseMapper: searchResponseMapper
+});
+```
+
+### Implement your own adapter using schemas
+
+The `x-adapter` library includes a `schemaMapperFactory` function that creates a mapper function to
+execute the schema you provide. You just need to create your request and response schemas to define
+the structure of your data, and pass them as a parameter to the factory function.
+
+```tsx
+import { schemaMapperFactory } from '@empathyco/x-adapter';
+import { SearchRequest } from '@empathyco/x-types';
+import { searchRequestSchema } from '../../schemas/requests/search-request.schema';
+
+// Import and use already existing types from your API source, or declare your own based on the response you get.
+interface ApiRequestModel {
+  q: string;
+  limit?: number;
+  offset?: number;
+}
+interface ApiResponseModel {
+  items: T[];
+  limit?: number;
+  offset?: number;
+  total: number;
+}
+
+// Maps both the request and the response to connect your model with API you are working with.
+const searchRequestSchema: Schema<SearchRequest, ApiRequestModel> = {
+  q: 'query',
+  limit: 'rows',
+  offset: 'start'
+};
+const searchResponseSchema: Schema<ApiResponseModel, SearchResponse> = {
+  results: 'items',
+  totalResults: 'items.total' // you can use paths to access to an inner property if needed
+};
+
+// The 'schemaMapperFactory' creates a mapper function for a given schema.
+export const searchRequestMapper = schemaMapperFactory<SearchRequest, ApiRequestModel>(
+  searchRequestSchema
+);
+export const searchResponseMapper = schemaMapperFactory<ApiResponseModel, SearchResponse>(
+  searchResponseSchema
+);
+```
+
+#### Create more complex models with Subschemas
+
+For more complex data mapping, the `x-adapter` library also provides a `subSchema` type that
+receives a `Source`, a `Target` and a `Path` that will be applied to the inner path of the main
+object. Furthermore, and not restricted to using subschemas, you can also use schema with function
+return types, that makes schemas powerful to custom the logic you need in your data mapping.
+
+```tsx
+// Example of using a subSchema to map a response
+const responseMapper = schemaMapperFactory<ApiResponseModel, SearchResponse>({
+  totalResults: 'results.total',
+  results: {
+    $path: 'results.items',
+    $subSchema: {
+      id: 'id',
+      name: 'name',
+      url: 'result_url',
+      images: ({ result }) => result.images.map(({ url }) => url),
+      price: {
+        value: ({ price, sale_price }) => (sale_price ? sale_price : price)
+      }
+    }
+  }
+});
+```
+
+#### Using a mutable schema
+
+The `x-adapter` library also provides a `MutableSchema` type, a `Schema` that can replace an
+original schema, override it or create a new one based on the original one. This can be useful to
+define a shared pattern that you will need to reuse in different endpoint adapters.
```

Comment:
I wouldn't say that "a `MutableSchema` type, a `Schema` that can replace an
original schema, override it". 

> With the `x-adapter` you can create mutable schemas. These schemas are helpful when writing libraries where you want to have some defaults, but allow to modify or extend them.

I would also add an example for each one of the options that the mutable schemas has.